spec/requests/users_request_spec.rb

```diff
@@ -1,7 +1,28 @@
 require 'rails_helper'
 
 RSpec.describe 'Users', type: :request do
-  it 'should be a place holder test' do
-    expect('hello').to eq('hello')
+  it 'should succesfully request for users ' do
+    get '/users/'
+    expect(response.status).to eq(200)
+  end
+  it 'should succesfully render index template ' do
+    get '/users/'
+    expect(response.status).to render_template(:index)
+  end
+  it 'check the response body includes correct placeholder text' do
+    get '/users/'
+    expect(response.body).to include('<h1>All users place holder</h1>')
+  end
+  it 'should succesfully render show template ' do
+    get '/users/1'
+    expect(response.status).to render_template(:show)
+  end
+  it 'should succesfully request for users ' do
+    get '/users/1'
+    expect(response.status).to eq(200)
+  end
+  it 'check the response body includes correct placeholder text' do
+    get '/users/1'
+    expect(response.body).to include('<h1>user show placeholder</h1>')
   end
```

Comment:
- [ ] Kindly separate the index and show tests into different describe blocks for better code organization.

  ```
  describe 'GET /index' do
    ...
  end

  describe 'GET /show' do
    ...
  end
  ```